transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/backends/TransportBackend.java

```diff
@@ -15,9 +15,20 @@
 package com.google.android.datatransport.runtime.backends;
 
 import com.google.android.datatransport.runtime.EventInternal;
+import com.google.android.datatransport.runtime.TransportContext;
 
 public interface TransportBackend {
   EventInternal decorate(EventInternal event);
 
   BackendResponse send(BackendRequest backendRequest);
+
+  /**
+   * This method indicate whether data sent through {@link TransportBackend} with {@link
+   * TransportContext} should be recorded as client health metrics and upload to Flg server.
+   *
+   * <p>By default, it returns false, and events will not be recorded and uploaded to Flg server.
+   */
+  default boolean shouldUploadClientHealthMetrics(TransportContext transportContext) {
```

Comment:
To future proof this, a few  suggestions:

1. let's not return a plain boolean but rather return an `UploadOptions` object that has a boolean property
2. let's name the method a bit more generically, i.e.

```java
default UploadOptions getUploadOptions(TransportContext transportContext) {
  return UploadOptions.none();
}
```

The thinking behind it is that it will make it easier to add future `options`. i.e. we won't have introduce new methods to this interface but just add more properties to UploadOptions.

Reply:
Great point! It definitely makes it more scalable.